src/core/syscalls/syscall_request.rs

```diff
@@ -8,6 +8,8 @@ use felt::Felt252;
 #[derive(Debug, PartialEq)]
 pub(crate) enum SyscallRequest {
     EmitEvent(EmitEventStruct),
+    // Deprecated SyscallRequests variants
+    DeprecatedEmitEvent(DeprecatedEmitEventStruct),
```

Comment:
I think we might want to rename this file to `deprecated_syscall_request.rs` (and rename all structs accordingly) and have the newer ones be in `syscall_request.rs`. I imagine it will get very lengthy once all the new structs get implemented. 
We could also defer it to a later work item but since this might be the first PR we merge it seems like we could enforce it here.